### PR TITLE
Bug/64781 delete ps report staging file

### PR DESCRIPTION
**Pull request by jon-shipley**
Fix ps report gen

It now uses the service bus admin client to query the number of active messages on the queue and sends the staging start message when processing the last message.

### CODE REVIEW

File: tslib/src/functions-ps-report/ps-report-4-writer/index.ts [hunk 1]
```diff
@@ -38,8 +38,7 @@ async function bulkUpload (context: Context, incomingMessage: PsReportStagingCom
       await jobDataService.setJobComplete(incomingMessage.jobUuid,
         JobStatusCode.Failed, JSON.stringify(error))
     }
-    // Live debugging: 2024-05-31 for PS report. To be re-enabled.
-    // await service.cleanup(incomingMessage.filename)
+    await service.cleanup(incomingMessage.filename)
```
Comment (GuyHarwood) [inline tslib/src/functions-ps-report/ps-report-4-writer/index.ts:41]: let's leave the file intact until we are happy that everything is working correctly

File: tslib/src/functions-ps-report/ps-report-2-pupil-data/ps-report.service.ts [hunk 2]
```diff
@@ -65,28 +70,76 @@ export class PsReportService {
...
+    // See how many message are left on the "schools" sb queue
+    const inputQueueName = 'ps-report-schools'
+    const queueRuntimeProperties = await this.serviceBusAdministrationClient.getQueueRuntimeProperties(inputQueueName)
+
+    /**
+     * The number of messages left of the queue at which point we want to tell the staging-service to start running.
+     */
+
+    // const targetLowMessageCount = Math.ceil((config.PsReport.ListSchools.PercentLeftToStartStaging / 100) * incomingMessage.totalNumberOfSchools)
+    const msgCount = queueRuntimeProperties.totalMessageCount
```
Comment (GuyHarwood) [inline tslib/src/functions-ps-report/ps-report-2-pupil-data/ps-report.service.ts:109]: please use activeMessageCount, as total could include dead message counts

**Review by GuyHarwood — CHANGES_REQUESTED** [review]
just a couple of changes needed...

File: tslib/src/functions-ps-report/ps-report-4-writer/index.ts (hunk 1 above)
Comment (jon-shipley) [reply tslib/src/functions-ps-report/ps-report-4-writer/index.ts]: Done

File: tslib/src/functions-ps-report/ps-report-2-pupil-data/ps-report.service.ts (hunk 2 above)
Comment (jon-shipley) [reply tslib/src/functions-ps-report/ps-report-2-pupil-data/ps-report.service.ts]: Done

**Review by GuyHarwood — APPROVED** [review]
LGTM

**Pull request merged**